src/System.Runtime.Extensions/src/System/ApplicationId.cs

```diff
@@ -64,30 +72,26 @@ public override string ToString ()
                 sb.Append(ProcessorArchitecture);
                 sb.Append('"');
             }
-            return StringBuilderCache.GetStringAndRelease(sb);
+            return sb.ToString();
         }
 
         private static char HexDigit(int num) =>
             (char)((num < 10) ? (num + '0') : (num + ('A' - 10)));
         
         private static string EncodeHexString(byte[] sArray) 
         {
-            string result = null;
-    
-            if (sArray != null)
-            {
-                char[] hexOrder = new char[sArray.Length * 2];
-            
-                int digit;
-                for(int i = 0, j = 0; i < sArray.Length; i++) {
-                    digit = (int)((sArray[i] & 0xf0) >> 4);
-                    hexOrder[j++] = HexDigit(digit);
-                    digit = (int)(sArray[i] & 0x0f);
-                    hexOrder[j++] = HexDigit(digit);
-                }
-                result = new string(hexOrder);
+            if (sArray == null) return null;
+
+            Span<char> hexOrder = stackalloc char[sArray.Length * 2];
+
+            for(int i = 0, j = 0; i < sArray.Length; i++) {
+                int digit = (int)((sArray[i] & 0xf0) >> 4);
+                hexOrder[j++] = HexDigit(digit);
+
+                digit = (int)(sArray[i] & 0x0f);
+                hexOrder[j++] = HexDigit(digit);
             }
-            return result;
+            return new string(hexOrder);
```

Comment:
If you know the exact number of chars you can use the new string.Create overload with Span which is the absolute fastest way to build a string.

Reply:
Are you talking about 
`public static string Create<TState>(int length, TState state, SpanAction<char, TState> action)` ?
How I can use it — I couldn't find any clue for now...

Reply:
From unit test:
```
char[] input = expected.ToCharArray();
            string result = string.Create(input.Length, input, (span, state) =>
            {
                Assert.Same(input, state);
                for (int i = 0; i < state.Length; i++)
                {
                    span[i] = state[i];
                }
            });
```
That's really the faster than new string(span) ??

Reply:
Yes, that's the fastest way to create a string. For more info around that read the midsection about Span in the Runtime from @stephentoub: https://msdn.microsoft.com/en-us/magazine/mt814808.aspx

Reply:
Of course without creating the char array from the unit test... You just need the length, you want to avoid heap allocations where possible

Reply:
Changes pushed, thanks for enlightening. 
Am I right that we are relying that JIT will optimize out allocation of lambda here?

Reply:
cc @stephentoub for delegate costs.